### ISSUE TITLE
Feature Request - Icebreaker break ICE button

### ISSUE THREAD

**Issue by MasterAir** [mtgred/netrunner#259]
Great work. This is really looking good.

One thing I thought might be useful and not too onerous to code in is a 'break all subroutines' button for Icebreakers with multiple paid abilities. If you're breaking curtain wall with corroder (something that shouldn't be done too often but is occasionally advisable) it takes 22 mouse clicks. A button that matches strength and breaks all subroutines of the Ice being encountered (and tells you how much it costs) would be really good.

**Comment by mtgred** [comment]
I like the idea and will try to code it this week.

**Comment by nealterrell** [comment]
This would be a direct nerf to Curtain Wall -- inducing carpal tunnel is one of its most important uses :+1:

**Comment by mtgred** [comment]
Neal, you want to do it?

I just realised i don't know how the ICE strength code works while working on Patch.

BTW, the commit with the hosting support for Operations has just been pushed. Patch doesn't seems to boost strength though. Can you have a look at it?

**Comment by nealterrell** [comment]
I can look at Patch.

This issue could be a tricky implementation, though. I don't see there being enough information in the current icebreaker code to figure this out at runtime... but maybe I'm overlooking some things.

```clojure
{:abilities [
  {:cost [:credit 1] :msg "break 1 code gate subroutine"}
  {:cost [:credit 1] :msg "add 1 strength" :effect (effect (pump card 1))}]}
```

how would you examine that structure to and determine the cost to break a strength 5 code gate? OK, the first ability is consistently an ability to break a subroutine, and it has a cost... but there's no non-text data about how many subroutines are broken. (Can't just parse it, either... see Switchblade). The pump effect has a number to pump by, but it's not true that the second ability of all icebreakers is a pump effect (Deus Ex, fixed-strength). I might be able to figure out if the :effect of an ability is calling the `pump` function, but I don't know if that is a good long-term solution.

This feels like a lot of work, needing a solid design to be drafted, but maybe I'm not thinking Clojure-ly.

**Comment by nealterrell** [comment]
Also, ideally this would display the cost of the automation in the click menu of the icebreaker, but to the best of my knowledge, labels on abilities are not dynamic -- they are set once and never updated. So I don't know how I could make the dynamic cost display work.

**Comment by MasterAir** [comment]
Sorry, I had no idea how the ice was implemented when i suggested this. An alternative improvement to the UX that has a similar function would be to implement an input field where you could ask for a certain effect to be applied an integer number of times.

So in the example of corroder vs exterior curtain wall - select 9 and then click boost strength then select 3 and click break barrier subroutine. Less automation, but probably about as useful in this case. Also, the do 1 net damage on Komainu and Tsurugi could have the same function.

**Comment by nealterrell** [comment]
I submitted the first step in making this dream come true in #298. It's a baby step but necessary. I'll keep you posted with progress.

**Comment by nealterrell** [comment]
Update...

![curtain1](https://cloud.githubusercontent.com/assets/10083341/8505138/b154cbce-2193-11e5-9dcd-b419abbbbf52.PNG)
![curtain2](https://cloud.githubusercontent.com/assets/10083341/8505139/b155d2d0-2193-11e5-8e57-156f7d08d976.PNG)

**Comment by nealterrell** [comment]
Merged in #405!
